test/unit/functions/functions-api-client-internal.spec.ts

```diff
@@ -137,63 +138,72 @@ describe('FunctionsApiClient', () => {
     
     for (const invalidName of [null, NaN, 0, 1, true, false, '', [], [1, 'a'], {}, { a: 1 }, _.noop, undefined]) {
       it(`should throw if functionName is ${invalidName}`, () => {
-        expect(() => apiClient.enqueue({}, invalidName as any))
-          .to.throw('Function name must be a non empty string');
+        expect(apiClient.enqueue({}, invalidName as any))
+          .to.eventually.throw('Function name must be a non empty string');
       });
     }
 
     for (const invalidName of ['project/abc/locations/east/fname', 'location/west/', '//']) {
       it(`should throw if functionName is ${invalidName}`, () => {
-        expect(() => apiClient.enqueue({}, invalidName as any))
-          .to.throw('Function name must be a single string or a qualified resource name');
+        expect(apiClient.enqueue({}, invalidName as any))
+          .to.eventually.throw('Function name must be a single string or a qualified resource name');
       });
     }
 
     for (const invalidOption of [null, 'abc', '', [], true, 102, 1.2]) {
       it(`should throw if options is ${invalidOption}`, () => {
-        expect(() => apiClient.enqueue({}, FUNCTION_NAME, '', invalidOption as any))
-          .to.throw('TaskOptions must be a non-null object');
+        expect(apiClient.enqueue({}, FUNCTION_NAME, '', invalidOption as any))
```

Comment:
Hm I deliberately chose to use chai's `eventually` to wait on the Promise returned by `enqueue` calls instead of async/await, to align better with the existing implementation which generally avoids async/await. Should I go ahead and refactor the unit tests to use async/await instead?